### DISCUSSION TITLE
Desctop notifications

### DISCUSSION THREAD

**Discussion by JekRock** [starship/starship#3333] [Q&A] [answered]
Thanks for the cool prompt!

What is the preferred way of installing `starship` with desktop notification support for the `cmd_duration` module?

**Comment by chipbuster** [comment, marked as answer]
Right now, you're likely going to have to build it from source. The procedure will be something like this:

- Install rustup and the default rust toolchain from https://rustup.rs/
- Get the source code for starship, either via git or by a download from the release page.
- Execute `cargo build --features notify-rust --release` from within the source directory.
- This should give you a binary at `<source_dir>/target/release/starship` which is built with notify-rs support, which you can then install.

Of course, there are several ways this can screw up (e.g. you might need openssl or libdbus support), so feel free to post if you hit any problems.

**Reply by JekRock** [reply]
Thanks!

Building from sources is less convenient for me as I'll have to keep track of new versions and build them manually. So I think I'll continue using `starship` without notification support for now.
